src/inspector/inspector_agent.cpp

```diff
@@ -405,20 +374,23 @@ void InspectorWrapConsoleCall(const v8::FunctionCallbackInfo<v8::Value> &args) {
 }
 
 void AgentImpl::Start() {
-  std::thread([this]() {
-    InspectorAgentDelegate delegate(this, "", script_name_, wait_);
-    delegate_ = &delegate;
+  auto self(shared_from_this());
+  std::thread([this, self]() {

```

Comment:
>this, self] [](start = 15, length = 11)

(nit) Is that a good pattern to pass both "this" and "self"? why not just pass "self" and use it in the function?

Reply:
It is pretty strange, and needs to be followed up on, but I copied existing code in the inspector that does the same thing.

---
In reply to: [447994164](https://github.com/microsoft/v8-jsi/pull/17#discussion_r447994164) [](ancestors = 447994164)